SpotifyUserLogin/src/main/java/main/Main.java

```diff
@@ -1,82 +1,135 @@
+        profileView.add(artistsLabel);
+
+        JButton backButton = new JButton("Back to Main Menu");
+        profileView.add(backButton);
+
         frame.getContentPane().add(mainMenuView, "MainMenu");
-        frame.getContentPane().add(dashboardView.getPanel(), "Dashboard");
         frame.getContentPane().add(profileView, "Profile");
-        frame.getContentPane().add(playlistView, "Playlist");
-        frame.getContentPane().add(rateCommentView, "RateComment");
 
-        // Access CardLayout for navigation
         CardLayout cardLayout = (CardLayout) frame.getContentPane().getLayout();
 
-        // MainMenu navigation
-        mainMenuView.addProfileButtonListener(e -> {
-            profileView.displayUserProfile(userProfile);
-            cardLayout.show(frame.getContentPane(), "Profile");
-        });
-
-        mainMenuView.addSearchFriendsButtonListener(e -> cardLayout.show(frame.getContentPane(), "Dashboard"));
+        // MainMenu -> Profile Navigation
+        profileButton.addActionListener(e -> {
+            // Fetch and display preferences
+            controller.updatePreferencesDynamically();
+            EditPreferencesState state = controller.getState();
 
-        // Profile navigation
-        profileView.addBackButtonListener(e -> cardLayout.show(frame.getContentPane(), "MainMenu"));
-        profileView.addPlaylistButtonListener(e -> {
-            playlistView.displayPlaylist(userProfile.getPlaylist().get(0)); // Display the first playlist
-            cardLayout.show(frame.getContentPane(), "Playlist");
+            if (state != null && state.isSuccess()) {
+                usernameLabel.setText("Username: " + userProfile.getUsername());
+                genresLabel.setText("Preferred Genres: " + String.join(", ", state.getGenres()));
+                artistsLabel.setText("Preferred Artists: " + String.join(", ", state.getArtists()));
+                cardLayout.show(frame.getContentPane(), "Profile");
+            } else {
+                JOptionPane.showMessageDialog(frame, "Error updating preferences: " + (state != null ? state.getMessage() : "Unknown error"), "Error", JOptionPane.ERROR_MESSAGE);
+            }
         });
-        profileView.addRatingsButtonListener(e -> cardLayout.show(frame.getContentPane(), "RateComment"));
 
-        // Playlist back navigation
-        playlistView.addBackButtonListener(e -> cardLayout.show(frame.getContentPane(), "Profile"));
+        // Profile -> MainMenu Navigation
+        backButton.addActionListener(e -> cardLayout.show(frame.getContentPane(), "MainMenu"));
 
-        // RateComment back navigation
-        rateCommentView.addBackButtonListener(e -> cardLayout.show(frame.getContentPane(), "Profile"));
-
-        // Show the MainMenu view initially
+        // Display the main frame
         frame.setLocationRelativeTo(null);
         frame.setVisible(true);
     }
+
+    private static boolean authenticate(SpotifyInteractor interactor) {
```

Comment:
Much of this authentication program has already been written in the SpotifyInteractor class. Try to use that to avoid duplicated code